src/Task/Infection.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Infection task
+ */
+class Infection extends AbstractExternalTask
+{
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'infection';
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+
+        $resolver->setDefaults([
+            'threads' => null,
+            'test_framework' => null,
+            'only_covered' => null,
+            'configuration' => null,
+            'min_msi' => null,
+            'min_covered_msi' => null,
+            'mutators' => null,
+            'triggered_by' => ['php'],
+        ]);
+
+        $resolver->addAllowedTypes('threads', ['null', 'int']);
+        $resolver->addAllowedTypes('test_framework', ['null', 'string']);
+        $resolver->addAllowedTypes('only_covered', ['null', 'bool']);
+        $resolver->addAllowedTypes('configuration', ['null', 'string']);
+        $resolver->addAllowedTypes('min_msi', ['null', 'integer']);
+        $resolver->addAllowedTypes('min_covered_msi', ['null', 'integer']);
+        $resolver->addAllowedTypes('mutators', ['null', 'string']);
+        $resolver->addAllowedTypes('triggered_by', ['array']);
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run(ContextInterface $context)
+    {
+        $config = $this->getConfiguration();
+        $files = $context->getFiles()->extensions($config['triggered_by']);
+
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('infection');
+
+        $arguments->addOptionalArgument('--threads=%s', $config['threads']);
+        $arguments->addOptionalArgument('--test-framework=%s', $config['test_framework']);
+        $arguments->addOptionalArgument('--only-covered', $config['only_covered']);
+        $arguments->addOptionalArgument('--configuration=%s', $config['configuration']);
```

Comment:
This option is not documented: https://infection.github.io/guide/command-line-options.html
Does it work?

Reply:
Documentation updated